### PR TITLE
Feature better table

### PR DESCRIPTION
**Pull request by grburgess**
This addresses #154 A new table model format is created that does not rely on pandas and a t.clean() member function is created that removes the interpolators when creating many table models. 

A function to convert old table models from their format into the new one is created. 

This frees us from relying on pandas to keep its HDF5 format stable. 

This must be merged after #153  as I worked off this branch.

### CODE REVIEW

File: astromodels/functions/template_model.py [hunk 1]
```diff
@@ -428,68 +518,64 @@ def _custom_init_(self, model_name, other_name=None, log_interp=True):
...
+                k = key
+                                
+            log.debug(f"reading parameter {str(k)}")
 
-                    processed_parameters += 1
+  
+            self._parameters_grids[str(k)] = template_file.parameters[key]
 
-            self._energies = np.array(store["energies"])
+            processed_parameters += 1
```
Comment (BjoernBiltzinger) [inline astromodels/functions/template_model.py:567]: I think you can delete this processed_parameters variable? I don't see where it is used.

File: astromodels/functions/template_model.py [hunk 2]
```diff
@@ -834,3 +952,106 @@ def to_table_model(self, file_name, model_name, overwrite=False):
...
+                continue
+            
+            else:
+                
+                tokens = match.groups()
+                
+                this_parameter_number = int(tokens[0])
+                this_parameter_name = str(tokens[1])
+                
+                assert (
```
Comment (BjoernBiltzinger) [inline astromodels/functions/template_model.py:1000]: maybe change to log.error and AssertionError?

File: astromodels/tests/test_template_model.py [hunk 3]
```diff
@@ -86,6 +92,8 @@ def test_template_factory():
 
     tm(energies)
 
+    tm.clean()
+
```
Comment (BjoernBiltzinger) [inline astromodels/tests/test_template_model.py:99]: Maybe add one more test to test the conversion from old table model to new one? Without that future changes in the table model may break this conversion without us noticing.

**Review by BjoernBiltzinger — COMMENTED** [review]
Looks good in general. I only have three short comments.

File: astromodels/functions/template_model.py (hunk 2 above)
Comment (grburgess) [reply astromodels/functions/template_model.py:1000]: done!

File: astromodels/functions/template_model.py (hunk 1 above)
Comment (grburgess) [reply astromodels/functions/template_model.py:567]: done!

File: astromodels/tests/test_template_model.py (hunk 3 above)
Comment (grburgess) [reply astromodels/tests/test_template_model.py:99]: done!

**Review by BjoernBiltzinger — APPROVED** [review]
Very nice

**Pull request merged**